extensions/amp-mega-menu/0.1/test/test-amp-mega-menu.js

```diff
@@ -0,0 +1,177 @@
+      heading.dispatchEvent(clickEvent);
+      expect(heading.parentNode.hasAttribute('open')).to.be.false;
+      expect(heading.getAttribute('aria-expanded')).to.equal('false');
+    });
+
+    it('should collapse an expanded menu item when another heading is clicked', () => {
+      element.build();
+      const heading1 = doc.getElementById('heading1');
+      const heading2 = doc.getElementById('heading2');
+      element.implementation_.expand_(heading1.parentNode);
+      expect(heading1.parentNode.hasAttribute('open')).to.be.true;
+      expect(heading1.getAttribute('aria-expanded')).to.equal('true');
+      const clickEvent = new Event('click');
+      heading2.dispatchEvent(clickEvent);
+      expect(heading1.parentNode.hasAttribute('open')).to.be.false;
+      expect(heading1.getAttribute('aria-expanded')).to.equal('false');
+      expect(heading2.parentNode.hasAttribute('open')).to.be.true;
+      expect(heading2.getAttribute('aria-expanded')).to.equal('true');
+    });
+
+    it('should collapse any expanded item after clicking outside the component', () => {
+      element.build();
+      const heading = doc.getElementById('heading1');
+      element.implementation_.expand_(heading.parentNode);
+      expect(heading.parentNode.hasAttribute('open')).to.be.true;
+      expect(heading.getAttribute('aria-expanded')).to.equal('true');
+      const clickEvent = new Event('click');
+      doc.documentElement.dispatchEvent(clickEvent);
+      expect(heading.parentNode.hasAttribute('open')).to.be.false;
+      expect(heading.getAttribute('aria-expanded')).to.equal('false');
+    });
+
+    it('should not collapse when click is inside the expanded content', () => {
+      element.build();
+      const heading = doc.getElementById('heading1');
+      const content = doc.getElementById('content1');
+      element.implementation_.expand_(heading.parentNode);
+      expect(heading.parentNode.hasAttribute('open')).to.be.true;
+      expect(heading.getAttribute('aria-expanded')).to.equal('true');
+      const clickEvent = new Event('click');
+      content.dispatchEvent(clickEvent);
+      expect(heading.parentNode.hasAttribute('open')).to.be.true;
+      expect(heading.getAttribute('aria-expanded')).to.equal('true');
+    });
+
+    it('should collapse when ESC key is pressed', () => {
+      element.build();
+      const heading = doc.getElementById('heading1');
+      element.implementation_.expand_(heading.parentNode);
+      expect(heading.parentNode.hasAttribute('open')).to.be.true;
+      expect(heading.getAttribute('aria-expanded')).to.equal('true');
+      const escKey = new KeyboardEvent('keydown', {key: Keys.ESCAPE});
+      doc.documentElement.dispatchEvent(escKey);
+      expect(heading.parentNode.hasAttribute('open')).to.be.false;
+      expect(heading.getAttribute('aria-expanded')).to.equal('false');
+    });
+
+    it('should be navigable by left/right arrow keys when a heading has focus', () => {
```

Comment:
We should use `async` `await` here to wait for the promise to return. E.g.: 

```
async () => {
...
await promise; 
} 
```

Reply:
Should the other tests in this unit test suite also be waiting for build callback to complete?

Reply:
Adding onto cathyxz, I believe tests that run asynchronous callbacks run the risk of executing and, if the promise fails, never calling the `expect`s. IIRC there are two ways around this, either with a callback function ie.

```javascript
it('..', async done => {
  ...
  done();
}
```

or by returning a promise, in which case Mocha waits for the promise to resolve.